presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -502,6 +503,23 @@ private static RowExpression changeType(RowExpression value, Type targetType)
             return value.accept(visitor, null);
         }
 
+        private static boolean allChildrenTypeOnly(Cast node)
+        {
+            // TODO this is a hack to prevent accidentally change children type if children of a cast is not type only cast
+            ImmutableList.Builder<Boolean> casts = ImmutableList.builder();
+            new DefaultExpressionTraversalVisitor<Void, ImmutableList.Builder<Boolean>>()
+            {
+                @Override
+                protected Void visitCast(Cast node, ImmutableList.Builder<Boolean> context)
+                {
+                    super.visitCast(node, context);
+                    context.add(node.isTypeOnly());
```

Comment:
You don't really need a list here. You can just update it as `context = context && node.isTypeOnly()` assuming context is `Boolean`.

Reply:
Ah, I was thinking what will happen when the list is empty but yes, we can use an settable boolean.